### PR TITLE
Filter out `Document`s without text in `DocumentEmbedder` + tests

### PR DESCRIPTION
**Pull request by OskarStark**

### CODE REVIEW

File: tests/DocumentEmbedderTest.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+use PhpLlm\LlmChain\Document\Vector;
+use PhpLlm\LlmChain\DocumentEmbedder;
+use PhpLlm\LlmChain\EmbeddingsModel;
+use PhpLlm\LlmChain\Store\StoreInterface;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\Attributes\Test;
+use PHPUnit\Framework\TestCase;
+
+#[CoversClass(DocumentEmbedder::class)]
+final class DocumentEmbedderTest extends TestCase
```
Comment (chr-hertel) [inline tests/DocumentEmbedderTest.php:17]: with that level of mocking i would propose to consider it a unit test and add `#[Small]` attribute here

File: tests/DocumentEmbedderTest.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    protected function setUp(): void
+    {
+        $this->embeddings = $this->createMock(EmbeddingsModel::class);
+        $this->store = $this->createMock(StoreInterface::class);
+        $this->embedder = new DocumentEmbedder($this->embeddings, $this->store);
+    }
+
+    #[Test]
+    public function testEmbedSingleDocument(): void
```
Comment (chr-hertel) [inline tests/DocumentEmbedderTest.php:31]: i guess gpt was not aware the `test*` prefix is not needed here? :D

Comment (OskarStark) [reply tests/DocumentEmbedderTest.php:36]: done

File: tests/DocumentEmbedderTest.php (hunk 1 above)
Comment (OskarStark) [reply tests/DocumentEmbedderTest.php:20]: done, not really used to this attributes, sorry

File: examples/store-mongodb-similarity-search.php [hunk 3]
```diff
@@ -55,7 +56,7 @@
 
 // create embeddings for documents
 $platform = new OpenAI(HttpClient::create(), $_ENV['OPENAI_API_KEY']);
-$embedder = new DocumentEmbedder($embeddings = new Embeddings($platform), $store);
+$embedder = new DocumentEmbedder($embeddings = new Embeddings($platform), $store, new NullLogger());
```
Comment (OskarStark) [inline examples/store-mongodb-similarity-search.php:59]: `NullLogger` is fine here?

File: tests/DocumentEmbedderTest.php [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+    //    #[Test]
+    //    public function embedDocumentWithoutText(): void
+    //    {
+    //        $document = new Document(Uuid::v4(), null, null);
+    //        $this->embeddings->expects($this->once())->method('multiCreate');
+    //        $this->store->expects($this->once())->method('addDocuments');
+    //
+    //        $this->embedder->embed($document);
+    //    }
```
Comment (OskarStark) [inline tests/DocumentEmbedderTest.php:71]: I don't think we want to can it if text is null, but that happens right now.

File: examples/store-mongodb-similarity-search.php (hunk 3 above)
Comment (chr-hertel) [reply examples/store-mongodb-similarity-search.php:59]: could also be part of the constructor as default, but don't really have a preference tbh

File: tests/DocumentEmbedderTest.php (hunk 4 above)
Comment (chr-hertel) [reply tests/DocumentEmbedderTest.php:71]: log and skip?

Comment (OskarStark) [reply tests/DocumentEmbedderTest.php:71]: that instead of an exception

Comment (OskarStark) [reply tests/DocumentEmbedderTest.php]: done

File: examples/store-mongodb-similarity-search.php (hunk 3 above)
Comment (OskarStark) [reply examples/store-mongodb-similarity-search.php:59]: will do

Comment (OskarStark) [reply examples/store-mongodb-similarity-search.php]: done

**Comment by OskarStark** [comment]
Ready to merge from my side

**Review by chr-hertel — APPROVED** [review]

**Pull request merged**